newrelic/resource_newrelic_workload.go

```diff
@@ -0,0 +1,327 @@
+package newrelic
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/newrelic/newrelic-client-go/pkg/errors"
+	"github.com/newrelic/newrelic-client-go/pkg/workloads"
+)
+
+func resourceNewRelicWorkload() *schema.Resource {
+	userReference := &schema.Schema{
+		Type:     schema.TypeList,
+		MaxItems: 1,
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"email": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"gravatar": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"id": {
+					Type:     schema.TypeInt,
+					Computed: true,
+				},
+				"name": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+			},
+		},
+	}
+
+	return &schema.Resource{
+		Create: resourceNewRelicWorkloadCreate,
+		Read:   resourceNewRelicWorkloadRead,
+		Update: resourceNewRelicWorkloadUpdate,
+		Delete: resourceNewRelicWorkloadDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"account": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				Computed:    true,
+				Description: "The account the workload belongs to.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
```

Comment:
Is `id` or `name` required here? You could add `ExactlyOneOf: []string{"account.0.id", "account.0.name"}` both here and in the `name` declaration to force that behavior.

Reply:
I've flattened this down to `account_id` which is required.